src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -109735,6 +109735,996 @@ class Program
             comp.VerifyDiagnostics();
         }
 
+        [Fact, WorkItem(35816, "https://github.com/dotnet/roslyn/issues/35816")]
+        public void StaticEqualsMethod_NonNullExpr_LiteralNull_WarnsWhenTrue()
+        {
+            var source0 = @"
+#nullable enable
+class C
+{
+    static void M(string s)
+    {
+        if (object.Equals(s, null))
+        {
+            _ = s.ToString(); // 1
+        }
+        else
+        {
+            _ = s.ToString();
+        }
+    }
+
+    static void M2(string s)
+    {
+        if (object.Equals(null, s))
+        {
+            _ = s.ToString(); // 2
+        }
+        else
+        {
+            _ = s.ToString();
+        }
+    }
+}";
+            verify(source0);
+            verify(source0.Replace("object.Equals", "object.ReferenceEquals"));
+
+            void verify(string source)
+            {
+                var comp = CreateCompilation(source);
+                comp.VerifyDiagnostics(
+                        // (9,17): warning CS8602: Dereference of a possibly null reference.
+                        //             _ = s.ToString(); // 1
+                        Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "s").WithLocation(9, 17),
+                        // (21,17): warning CS8602: Dereference of a possibly null reference.
+                        //             _ = s.ToString(); // 2
+                        Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "s").WithLocation(21, 17));
+            }
+        }
+
+        [Fact, WorkItem(35816, "https://github.com/dotnet/roslyn/issues/35816")]
+        public void StaticEqualsMethod_NonNullExpr_NonNullExpr_NoWarning()
+        {
+            var source0 = @"
+#nullable enable
+class C
+{
+    static void M(string s1, string s2)
+    {
+        if (object.Equals(s1, s2))
+        {
+            _ = s1.ToString();
+            _ = s2.ToString();
+        }
+        else
+        {
+            _ = s1.ToString();
+            _ = s2.ToString();
+        }
+    }
+
+    static void M2(string s1, string s2)
+    {
+        if (object.Equals(s2, s1))
+        {
+            _ = s1.ToString();
+            _ = s2.ToString();
+        }
+        else
+        {
+            _ = s1.ToString();
+            _ = s2.ToString();
+        }
+    }
+}";
+            verify(source0);
+            verify(source0.Replace("object.Equals", "object.ReferenceEquals"));
```

Comment:
Nit: consider using a `Theory` instead, and having the argument for the test be `Equals` and `ReferenceEquals`

Reply:
Same suggestion for the test of the tests that do this style of multi-verification

---
In reply to: [298742762](https://github.com/dotnet/roslyn/pull/36722#discussion_r298742762) [](ancestors = 298742762)